src/block.js

```diff
@@ -137,8 +138,131 @@ module.exports = (common) => {
           })
         })
       })
+
+      describe('.rm', function () {
+        it('by CID', (done) => {
+          const expectedHash = 'QmPv52ekjS75L4JmHpXVeuJ5uX2ecSfSZo88NSyxwA3rAQ'
+          const cid = new CID(expectedHash)
+          const b = new Block(new Buffer('blorb'), cid)
+
+          series([
+            (cb) => {
+              // Add a block
+              ipfs.block.put(b, (err, block) => {
+                expect(err).to.not.exist()
+                expect(block.data).to.be.eql(b.data)
+                expectKey(block, multihash.fromB58String(expectedHash), cb)
+              })
+            },
+            (cb) => {
+              // Remove the block
+              ipfs.block.rm(cid, (err) => {
+                expect(err).to.not.exist()
+                cb()
+              })
+            },
+            (cb) => {
+              // Remove bitswap from IPFS so that block.get tries to fetch the block
+              // from the local repo (not from the network). This is to make sure we
+              // get the error (as expected) instead of waiting for a timeout.
+              ipfs._blockService._bitswap = null
```

Comment:
The tests can't use any internals otherwise it will fail with js-ipfs-api

Reply:
Oh :/ I figured that was a a bit of a workaround. Any suggestions how to avoid the block.get() trying to get the block from bitswap thus causing it "never" to return? The idea here is to get block.get() to return an error (ie. "we don't have the block anymore").

Reply:
You will just have to .skip that test until we have https://github.com/ipfs/interface-ipfs-core/issues/58, unfortunately.

Reply:
Ok. That's unfortunate. I'll update the commit and .skip it for now. Let's leave the tests there though, so that we have them ready for the future.

Reply:
Committed. I removed the line where bitswap was nulled, added .skip to both callback and promise tests for .rm() and added some notes in the code as to why.